src/metatensor/models/cli/train.py

```diff
@@ -375,16 +376,22 @@ def _train_model_hydra(options: DictConfig) -> None:
         outputs=outputs,
     )
 
+    # process devices
+    architecture_devices = architecture.DEVICES
+    requested_device = options["device"]
+    available_devices = get_available_devices()
+    devices = pick_devices(requested_device, available_devices, architecture_devices)
```

Comment:
Can't this be just a single function I don't really the point of having four lines. This can even be called directly below when we pass it to the model or even within the training function at the beginning of the function.

```python
model = architecture.train(
   ...
   devices = resolve_device(
       requested_device=options["device"],
       available_devices=architecture.DEVICES,
)
```

Reply:
The issue with having it as a single function is that the single function becomes untestable, because we have no control over the available devices in a test environment. In this way, the `pick_devices` function is testable

Reply:
At the end, we return a list of `torch.device`s. If those cannot be created because they're not available, there is no way around it... even if you patch the tests

Reply:
make sense.